src/mkindex_algo.hpp

```diff
@@ -119,24 +120,23 @@ auto loadSubjSeqsAndIds(LambdaIndexerOptions const & options)
     double start = sysTime();
     myPrint(options, 1, "Loading Subject Sequences and Ids...");
 
-    using seq_traits = std::conditional_t<seqan3::nucleotide_alphabet<TOrigAlph>,
-                                          seqan3::sequence_file_input_default_traits_dna,
-                                          seqan3::sequence_file_input_default_traits_aa>;
-    seqan3::sequence_file_input<seq_traits, seqan3::fields<seqan3::field::id, seqan3::field::seq>> infile{
-      options.dbFile};
+    bio::io::seq::record r{.id   = std::string_view{},
+                           .seq  = bio::ranges::views::char_conversion_view_t<TOrigAlph>{},
+                           .qual = std::ignore};
+    bio::io::seq::reader reader{
+      options.dbFile,
+      bio::io::seq::reader_options{.record = r, .truncate_ids = options.truncateIDs}
+    };
 
     size_t count = 0;
-    for (auto & [id, seq] : infile)
+    for (auto & [id, seq, qual] : reader)
```

Comment:
Yes, the records in biocpp-io are just regular structs and always have a quality member. For FastA files, it will be empty. And it can be set to ignore to avoid reading it in FastQs.